### PR TITLE
Add Jetpack Cloud Activity Log

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Support `/backups/activity` route
* Add new `ActivityLog` component to be shared between filtered backup views and the activity page

Screen Shot 2020-03-27 at 2 56 07 PM

#### Testing instructions

1. Navigate to `/backups/activity` and verify after selecting a site you are taken to the activity log for that site
2. Reload the page and verify it takes you directly to the activity log for that site
3. Verify that the new activity log view works more or less the same as the filtered date view on the backups page.

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/backups/components/activity-log/style.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
+.activity-log {
+	// todo: there is some strangeness with the jetpack cloud css, these are temporary fixes
+	.filterbar__wrap.card {
```
Comment (cbauerman) [inline client/landing/jetpack-cloud/sections/backups/components/activity-log/style.scss]: as per the comments these rules get `ActivityLog` looking presentable but are a hack for figuring out what the heck is actually happening with some of these css rules

File: client/landing/jetpack-cloud/sections/backups/index.js [hunk 2]
```diff
@@ -11,16 +11,25 @@ import config from 'config';
...
+	backups,
 } from 'landing/jetpack-cloud/sections/backups/controller';
 
 export default function() {
 	if ( config.isEnabled( 'jetpack-cloud/backups' ) ) {
-		page( '/backups', siteSelection, sites, navigation, makeLayout, clientRender );
-		page( '/backups/:site', siteSelection, navigation, backups, makeLayout, clientRender );
+		page( '/backups/activity', siteSelection, sites, makeLayout, clientRender );
+		page(
+			'/backups/activity/:site',
```
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/index.js:25]: I prefer to follow the same route pattern as the rest pages `/backups/:site/{sub-section}`. When we merge your proposal on https://github.com/Automattic/wp-calypso/pull/40630 then it will follow this pattern with its own path function.

File: client/state/data-getters/index.js [hunk 3]
```diff
@@ -61,17 +61,21 @@ export const requestActivityActionTypeCounts = (
 	);
 };
 
-export const requestActivityLogs = ( siteId, filter, { freshness = 5 * 60 * 1000 } = {} ) => {
+export const requestActivityLogsId = ( siteId, filter ) => {
```
Comment (cleacos) [inline client/state/data-getters/index.js:64]: Why do we need to export this function? It seems like a 'private' helper function to create only the Id for the API.

File: client/landing/jetpack-cloud/sections/backups/activity/index.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import BackupActivityLog from 'landing/jetpack-cloud/sections/backups/components/activity-log';
+import DocumentHead from 'components/data/document-head';
+import Main from 'components/main';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
+
+const BackupActivityPage: FunctionComponent = () => {
```
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/activity/index.tsx:17]: Name suggestion `BackupActivityLogPage` to use the same naming as the design and to differentiate it from the `BackupActivityLog` component

image

File: client/landing/jetpack-cloud/sections/backups/components/activity-log/index.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	return (
+		<div className="activity-log">
+			<Filterbar
+				{ ...{
+					siteId,
+					filter,
+					isVisible: true,
+					isLoading: ! activities,
+				} }
+			/>
```
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/components/activity-log/index.tsx:117]: Could we move the Filter bar to the upper component `BackupActivityPage`? This way we'll be able to use this component in other parts, on the `BackupPage`, on the `BackupDetailPage` and also on the `BackupActivityLogPage`

**Backup Page**
image

**Backup Detail Page**
image

**Backup Activity Log Page**
image

Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/components/activity-log/index.tsx:117]: My thinking was possibly allowing it to render some or all of the filter bar depending on the context controlled via props. I.e. `showActivityTypePicker` & `showDatePicker`, but let me iterate on that

File: client/landing/jetpack-cloud/sections/backups/index.js (hunk 2 above)
Comment (cbauerman) [reply client/landing/jetpack-cloud/sections/backups/index.js:25]: The reason why this route does not follow the sub section logic of the the sections is that it is possible to navigate here without a site ( from the sidebar, for example ) and that navigating here without a site makes sense ( since we don't need anymore information once we get here, as opposed to restore, download, or detail which all need another id to display anything )

With that stated,  [the `siteSelection` function we use from Calypso](https://github.com/Automattic/wp-calypso/blob/5f83029a8074db20672622920904ebee388aa914/client/my-sites/controller.js#L320) does not have sophisticated logic when a site is picked: [It is literally just added to the end up the existing path](https://github.com/Automattic/wp-calypso/blob/5f83029a8074db20672622920904ebee388aa914/client/my-sites/controller.js#L358).

So, to sum up, when using a route that is compatible with siteSelection we need to be able to just append the siteSlug to the end, hence the `backups/activity` approach

File: client/state/data-getters/index.js (hunk 3 above)
Comment (cbauerman) [reply client/state/data-getters/index.js:64]: [You can see it in use here](https://github.com/Automattic/wp-calypso/pull/40494/files#diff-90651e10bbc95d5c8b38fe246249c55fR53). To get access to this data we need the id, [which indexes the response in httpData map.](https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/http-data.ts)
